### PR TITLE
feat: Set default EKS to 1.26; install vpc-cni add-on

### PR DESCRIPTION
**Pull request by gls4**
This PR changes the default version of EKS to 1.26. It also installs the AWS VPC-CNI add-on, and creates an IAM role for that add-on. If you find appying fails because some of the add-ons already exist, remove the add-ons in question using the AWS console or the AWS CLI and re-apply.

### CODE REVIEW

File: modules/app_eks/add-ons.tf [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+  depends_on = [
+    aws_eks_addon.vpc-cni
+  ]
+  cluster_name                = var.namespace
+  addon_name                  = "coredns"
+  addon_version               = "v1.9.3-eksbuild.11"
+  resolve_conflicts           = "OVERWRITE"
+}
+
+resource "aws_eks_addon" "kube-proxy" {
```
Comment (jsbroks) [inline modules/app_eks/add-ons.tf:66]:
```suggestion
resource "aws_eks_addon" "kube_proxy" {
```

File: modules/app_eks/add-ons.tf [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  depends_on = [
+    aws_eks_addon.vpc-cni
+  ]
+  cluster_name                = var.namespace
+  addon_name                  = "kube-proxy"
+  addon_version               = "v1.25.14-eksbuild.2"
+  resolve_conflicts           = "OVERWRITE"
+}
+
+resource "aws_eks_addon" "vpc-cni" {
```
Comment (jsbroks) [inline modules/app_eks/add-ons.tf:76]:
```suggestion
resource "aws_eks_addon" "vpc_cni" {
```

File: modules/app_eks/add-ons.tf [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+}
+
+resource "aws_iam_role" "oidc" {
+  name               = join("-", [var.namespace, "oidc"])
+  assume_role_policy = data.aws_iam_policy_document.oidc_assume_role.json
+}
+
+
+
+
```
Comment (jsbroks) [inline modules/app_eks/add-ons.tf:33]:
```suggestion
```

File: modules/app_eks/add-ons.tf [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    }
+
+    principals {
+      identifiers = [aws_iam_openid_connect_provider.eks.arn]
+      type        = "Federated"
+    }
+  }
+}
+
+resource "aws_iam_role_policy_attachment" "eks-oidc" {
```
Comment (jsbroks) [inline modules/app_eks/add-ons.tf:21]:
```suggestion
resource "aws_iam_role_policy_attachment" "eks_oidc" {
```

File: modules/app_eks/add-ons.tf [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+   depends_on = [
+     aws_eks_addon.vpc-cni
+   ]
+   cluster_name               = var.namespace
+   addon_name                 = "aws-efs-csi-driver"
+   addon_version              = "v1.7.7-eksbuild.1"
+   resolve_conflicts          = "OVERWRITE"
+ }
+
+resource "aws_eks_addon" "aws-ebs-csi-driver" {
```
Comment (jsbroks) [inline modules/app_eks/add-ons.tf:46]:
```suggestion
resource "aws_eks_addon" "aws_ebs_csi_driver" {
```

File: modules/app_eks/add-ons.tf [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+  name               = join("-", [var.namespace, "oidc"])
+  assume_role_policy = data.aws_iam_policy_document.oidc_assume_role.json
+}
+
+
+
+
+
+### add-ons
+resource "aws_eks_addon" "aws-efs-csi-driver" {
```
Comment (jsbroks) [inline modules/app_eks/add-ons.tf:36]:
```suggestion
resource "aws_eks_addon" "aws_efs_csi-driver" {
```

File: modules/app_eks/add-ons.tf [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+  ]
+  cluster_name                = var.namespace
+  addon_name                  = "coredns"
+  addon_version               = "v1.9.3-eksbuild.11"
+  resolve_conflicts           = "OVERWRITE"
+}
+
+resource "aws_eks_addon" "kube-proxy" {
+  depends_on = [
+    aws_eks_addon.vpc-cni
```
Comment (jsbroks) [inline modules/app_eks/add-ons.tf:68]:
```suggestion
    aws_eks_addon.vpc_cni
```

File: modules/app_eks/add-ons.tf [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+  ]
+  cluster_name                = var.namespace
+  addon_name                  = "aws-ebs-csi-driver"
+  addon_version               = "v1.25.0-eksbuild.1"
+  resolve_conflicts           = "OVERWRITE"
+}
+
+resource "aws_eks_addon" "coredns" {
+  depends_on = [
+    aws_eks_addon.vpc-cni
```
Comment (jsbroks) [inline modules/app_eks/add-ons.tf:58]:
```suggestion
    aws_eks_addon.vpc_cni
```

File: modules/app_eks/add-ons.tf [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+   ]
+   cluster_name               = var.namespace
+   addon_name                 = "aws-efs-csi-driver"
+   addon_version              = "v1.7.7-eksbuild.1"
+   resolve_conflicts          = "OVERWRITE"
+ }
+
+resource "aws_eks_addon" "aws-ebs-csi-driver" {
+  depends_on = [
+    aws_eks_addon.vpc-cni
```
Comment (jsbroks) [inline modules/app_eks/add-ons.tf:48]:
```suggestion
    aws_eks_addon.vpc_cni
```

**Review by nfoucha — APPROVED** [review]

**Comment by jsbroks** [comment]
This PR is included in version 4.10.0 :tada:

**Pull request merged**